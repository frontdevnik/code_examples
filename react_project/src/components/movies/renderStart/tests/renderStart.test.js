import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import RenderStart from '../RenderStart';

describe('<RenderStart />', () => {
  let wrapper;
  const mockStore = configureStore();
  const initialState = {
    setRatingStars: jest.fn(),
    changeSelectedMovieStars: jest.fn(),
  }

  const props = {
    star: 5,
    movie: {},
    id: 2,
  }

  beforeEach(() => {
    const store = mockStore(initialState);
    wrapper = shallow(<RenderStart store={store} {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  
  it('should render the correct number of stars', () => {
    expect(wrapper.dive().find('div').length).toBe(5);
  });
});
