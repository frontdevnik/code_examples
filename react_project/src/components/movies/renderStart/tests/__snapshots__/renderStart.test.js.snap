// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RenderStart /> should render correctly 1`] = `"<div class=\\"starActive\\"></div><div class=\\"starActive\\"></div><div class=\\"starActive\\"></div><div class=\\"starActive\\"></div><div class=\\"starActive\\"></div>"`;
